### ISSUE TITLE
Stats file has a chance to corrupt on power outage when its being saved

### ISSUE THREAD

**Issue by viddie** [viddie/ConsistencyTrackerMod#92]
i could
- check if the file is corrupted. if yes: reset the stats file
- uhhh

```
like: when you begin your session, the stats file is saved in a different location as backup, and if a corrupt file is found cct can try to load the backup
then you would lose at most 1 session worth of stats
```

**Comment by viddie** [comment]
Different idea for this:

when writing the stats to the hard drive, dont write to the original stats file. instead write to `<original-file-name>.json.bak` and once fully written, move & replace the original stats file. on PC crash the .bak file will be corrupted, but the original file will still be there. the corrupt file will be overwritten when the stats file is written again.

**Comment by viddie** [comment]
Fixed as of d4150fe5e80b7a6f48a62e9390f9d5e9e2f8cdd9
